community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/staging/DynamicProcessorAssigner.java

```diff
@@ -98,15 +93,15 @@ private int assignProcessorsToPotentialBottleNeck( StageExecution execution, int
         if ( bottleNeck.other() > 1.0f &&
              batchesPassedSinceLastChange( bottleNeckStep, doneBatches ) >= config.movingAverageSize() )
         {
+            // Assign 1/10th of the remaining permits. This will have processors being assigned more
+            // aggressively in the beginning of the run
```

Comment:
It's not about 10. Say we have 100 permits. The first time 10 will be assigned, the next assignment will be 9 since there are now only 90 left. Then 8 and so forth. `optimalProcessorIncrement` is the optimal amount of processors to assign given the difference between the bottleneck step and the next-slowest. Of course the next slowest may also be parallelizable and so they would previously add one or two every now and then, slowly getting faster and faster than the other, so to reach optimal setup took a while. This speeds up the process of getting to the optimal setup, with a slight loss of perfect precision, which wasn't really necessary anyway.